File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.py
# Last updated: 6/4/2025, 9:24:07 PM
class Solution:
    def removeDuplicates(self, nums):
        if not nums:
            return 0
        
        k = 1  # Pointer for the position of unique elements
        
        for i in range(1, len(nums)):
            if nums[i] != nums[i - 1]:
                nums[k] = nums[i]
                k += 1
        
        return k

File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.java
// Last updated: 6/4/2025, 9:23:57 PM
class Solution {
    public int removeDuplicates(int[] nums) {
        int pointer1 = 1;
        for (int pointer2 = 1; pointer2<nums.length; pointer2++) {
            if (nums[pointer2] != nums[pointer2 - 1]) {
                nums[pointer1] = nums[pointer2];
                pointer1++;
            }
        }
        return pointer1;
    }
}